pilot/lib/pilot/tester_manager.rb

```diff
@@ -67,7 +67,14 @@ def remove_tester(options)
         # If no groups are passed to options, remove the tester from the app-level,
         # otherwise remove the tester from the groups specified.
         if config[:groups].nil? && tester.kind_of?(Spaceship::Tunes::Tester::External)
-          test_flight_tester = Spaceship::TestFlight::Tester.find(app_id: app.apple_id, email: tester.email)
+          test_flight_testers = Spaceship::TestFlight::Tester.search(app_id: app.apple_id, text: tester.email)
```

Comment:
Yes, this could totally happen. I was thinking we should rely on the caller to understand this (which is why I renamed it search, and added the comment about it being full-text). That being said... one thing I could do is add another optional param for "is_exact_email_match" or something like that, so it does that filtering you mention.